### PR TITLE
SG-1279 Malware Status

### PR DESCRIPTION
**Pull request by mdewolfe**
Adds handling for status codes returned when a file is pending, or failed, a malware scan.

### CODE REVIEW

File: shotgun_api3/shotgun.py [hunk 1]
```diff
@@ -2512,6 +2512,11 @@ def download_attachment(self, attachment=False, file_path=None, attachment_id=No
                             match = re.search('<Message>(.*)</Message>', xml)
                             if match:
                                 err += ' - %s' % (match.group(1))
+                elif e.code == 409 or e.code == 410:
+                    # we may be dealing with a file that is pending/failed a malware scan
+                    lines = e.readlines()
+                    if lines:
+                        err += "\n%s\n" % ''.join(lines)
```
Comment (mdewolfe) [inline shotgun_api3/shotgun.py]: modeled after error messages above

File: tests/test_api_long.py [hunk 2]
```diff
@@ -143,6 +145,72 @@ def test_schema_with_project(self):
             self.assertTrue('user' in schema)
             self.assertTrue('visible' in schema['user'])
 
+# Requires the Shotgun instance to have transcoding and 'transcoder_malware_support' enabled.
+class TestMalwareScanning(base.LiveTestBase):
```
Comment (mdewolfe) [inline tests/test_api_long.py]: Should these tests be in the workflow-tests?

Comment (daigles) [reply tests/test_api_long.py]: definitely! 
Lets move these tests to the workflow tests!

File: shotgun_api3/shotgun.py [hunk 3]
```diff
@@ -2512,6 +2512,11 @@ def download_attachment(self, attachment=False, file_path=None, attachment_id=No
                             match = re.search('<Message>(.*)</Message>', xml)
                             if match:
                                 err += ' - %s' % (match.group(1))
+                elif e.code == 409 or e.code == 410:
+                    # we may be dealing with a file that is pending/failed a malware scan
+                    lines = e.readlines()
```
[the remark this answers is not in the document]
Comment (mdewolfe) [reply shotgun_api3/shotgun.py]: Will do.

File: shotgun_api3/shotgun.py [hunk 4]
```diff
@@ -2512,6 +2512,11 @@ def download_attachment(self, attachment=False, file_path=None, attachment_id=No
                             match = re.search('<Message>(.*)</Message>', xml)
                             if match:
                                 err += ' - %s' % (match.group(1))
+                elif e.code == 409 or e.code == 410:
```
[the remark this answers is not in the document]
Comment (mdewolfe) [reply shotgun_api3/shotgun.py]: Will do

File: tests/test_api_long.py (hunk 2 above)
Comment (mdewolfe) [reply tests/test_api_long.py]: Will do.

**Comment by mdewolfe** [comment]
One outstanding issue, the `queue_for_s3_upload` field should be removed from the schema file. This is handled server side. https://github.com/shotgunsoftware/python-api/blob/master/tests/mockgun/schema.pickle#L10960

**Comment by mdewolfe** [comment]
@manneohrstrom PR updated with your and @daigles notes.

**Pull request merged**